File: App/RestClients/Microservices.Demo.RestClients/Pricing/Models/Requests/CalculatePriceRequest.cs
using Microservices.Demo.RestClients.Pricing.Models.Dtos;

namespace Microservices.Demo.RestClients.Pricing.Models.Requests
{
    public class CalculatePriceRequest
    {
        public string ProductCode { get; set; }
        public DateTimeOffset PolicyFrom { get; set; }
        public DateTimeOffset PolicyTo { get; set; }
        public List<string> SelectedCovers { get; set; }
        public List<QuestionAnswerDto> Answers { get; set; }
    }
}

File: App/Clients/ng-microservices-demo-client/src/app/infrastructure/rest-clients/policies/models/requests/calculate-price-request.ts
import { AutoMap } from "@automapper/classes";
import { QuestionAnswerDto } from "../dtos/question-answer-dto";

export class CalculatePriceRequest {
  @AutoMap()
  public productCode: string;
  @AutoMap()
  public policyFrom: Date;
  @AutoMap()
  public policyTo: Date;
  @AutoMap()
  public selectedCovers: string[];
  @AutoMap()
  public answers?: QuestionAnswerDto[];

  constructor(
    productCode: string = '',
    policyFrom: Date = new Date(),
    policyTo: Date = new Date(),
    selectedCovers: string[] = []
  ) {
    this.productCode = productCode;
    this.policyFrom = policyFrom;
    this.policyTo = policyTo;
    this.selectedCovers = selectedCovers;
  }
}
